astropy/stats/tests/test_funcs.py

```diff
@@ -242,6 +242,39 @@ def test_biweight_midvariance_axis_3d():
         bwi = np.array(bwi)
         assert_allclose(bw[y], bwi)
 
+def test_biweight_midcovariance():
+    """test biweight_midcovariance method"""
+    # Test 1
+    rng = np.random.RandomState(1)
+    d = np.array([rng.normal(0, 2, 100) for i in range(5)])
+    cov = funcs.biweight_midcovariance(d)
+    std = np.around(np.sqrt(cov.diagonal()), 1)
+    assert_allclose(std, [ 1.8,  1.9,  2. ,  2.1,  2.1])
+    # Test 2
+    rng = np.random.RandomState(1)
+    d = np.array([rng.normal(0, 2, 10) for i in range(5)])
+    cov = funcs.biweight_midcovariance(d)
+    std = np.around(np.sqrt(cov.diagonal()), 1)
+    assert_allclose(std, [ 2.6,  2.1,  1.6,  1.5,  1.9])
+
+def test_midcov_midvar():
+    """
+    test biweight_midcovariance and biweight_midvariance
+    are compatible
+    """
+    rng = np.random.RandomState(1)
+    d = np.array([rng.normal(0, 2, 100) for i in range(3)])
+    cov = funcs.biweight_midcovariance(d)
+    cov_std = np.around(np.sqrt(cov.diagonal()), 1)
+    std = np.around(list(map(funcs.biweight_midvariance, d)), 1)
+    assert_allclose(cov_std, std)
+
+def test_midcovariance_shape():
+    """
+    test that midcovariance raises error when feed 3D array
+    """
+    d = np.arange(27).reshape(3,3,3)
+    assert_raises(ValueError, funcs.biweight_midcovariance, d)
```

Comment:
We do not require nose for our tests, so please use ``pytest.raises`` instead. There are plenty of examples in this file.